### ISSUE TITLE
Orchestration function stuck in "Running status"

### ISSUE THREAD

**Issue by Haletski** [Azure/azure-functions-durable-extension#1510]
### Description
Orchestration instance is stuck in running status and not being processed further or executed after a long time. 

### Expected behavior
Suborchestation function should be executed.

### Actual behavior
Suborchetration function is stuck in running state

### App Details
- **Azure function version: 2**
- **Durable Functions extension version:  2.2.2**
- **Location: UK South**
- **App Name: func-dataforseo-dave-dev**
- **Orchetration Id d622ab47255c45f9a655716703689bf6**
- **.NET Core 3.1**:
- **Storage account: stdavemaindevenv**
- **Date: from 2020-10-05 10:04 UTC to 2020-10-05 12:38 UTC**

### Screenshots
![image](https://user-images.githubusercontent.com/30402551/95093106-13af9c80-0731-11eb-90b2-7a7404ccc12e.png)
![image](https://user-images.githubusercontent.com/30402551/95094012-178fee80-0732-11eb-8b04-3ebaa30cb543.png)

**Comment by cgillum** [comment]
Hi @Haletski. It looks like this instance is stuck waiting for a sub-orchestration to complete. The ID of the sub-orchestration is `30238f1589b04904a66c3d8fc18b2e06:1`. Looking at its behavior, it seems to be looping every minute or so and scheduling a bunch of activity functions in parallel.  Finally, at around 2020-10-05 12:38:46, the sub-orchestration appears to have gotten permanently stuck trying to load its state for the next timer message (I can see the dequeue count for this timer message has risen quite high) because of OutOfMemoryExceptions.

While I don't know for sure since I can't see your code, it looks to me like your sub-orchestration is written as a potentially infinite loop (or at least a very long loop). This is causing your sub-orchestration's history to grow very large and causing the framework to eventually fail to load it. I see you are on the Consumption plan and are therefore limited to 1.5 GB and likely exceeding this limit.

Two possible mitigations:

1. Rewrite your orchestration code to not loop so much, keeping in mind the guidance in our [Eternal orchestration documentation](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-eternal-orchestrations).
2. Upgrade your app to the Elastic Premium SKU where your memory limits are larger. Note that the current behavior is still dangerous because long history loads could potentially lead to other problems, but it would at least help this one orchestration instance progress a bit further.

Also, I see this orchestration instance is actually using v2.2.0 of the Durable Extension. I also recommend you upgrade to the latest version of the extension (v2.3.0 at the time of writing, but v2.3.1 will be out very soon) in order to get the latest reliability improvements.

**Comment by Haletski** [comment]
Hi @cgillum. Thank you so much for your detailed response.

You are absolutely correct that I'm using a very long loop. Please see below code snippet of sub-orchestration function.
    
          `
        [FunctionName("SerpKeywordsPaginatorOrchestrationFunction")]
        public static async Task Run([OrchestrationTrigger] IDurableOrchestrationContext context)
        {
            var getKeywordsInput = context.GetInput<Tuple<List<Keyword>, string>>();
            var requestPerMinuteCount = dataToPost.GetLimitCount(RequestsPerMinuteConstraint);
            var dataToPost = await context.CallActivityAsync<List<Data>>("SerpGetKeywordsActivityFunction", getKeywordsInput);
            var retryOptions = new RetryOptions(TimeSpan.FromSeconds(15), 3);

            for (var i = Zero; i < requestPerMinuteCount; i++)
            {
                var dataPerMinute = dataToPost
                    .Skip(i * RequestsPerMinuteConstraint)
                    .Take(RequestsPerMinuteConstraint)
                    .ToList();

                var requestCount = dataPerMinute.GetLimitCount(DataPerPostRequestConstraint);

                var requestPerMinuteTasks = Enumerable
                    .Range(Zero, requestCount)
                    .Select(x =>
                    {
                        var dataPerRequest = dataPerMinute
                            .Skip(x * DataPerPostRequestConstraint)
                            .Take(DataPerPostRequestConstraint);

                        var postKeywordsInput = Tuple.Create(dataPerRequest, getKeywordsInput.Item2);

                        return context.CallActivityWithRetryAsync<RootTaskExecution>("SerpPostKeywordsActivityFunction",
                            retryOptions, postKeywordsInput);
                    });

                await Task.WhenAll(requestPerMinuteTasks);

                var oneMinuteDelay = context.CurrentUtcDateTime.Add(TimeSpan.FromMinutes(1));

                await context.CreateTimer(oneMinuteDelay, CancellationToken.None);
            } 
        }
`
I understand that sub-orchestration function loop is quite big but it's needed because of business logic.

Is there any chance to decrease history of sub-orchestration function somehow? 

Or could you suggest something else for me as another option? (Elastic premium SKU will be not the best option for us).

Thank you in advance

**Comment by cgillum** [comment]
@Haletski the way to decrease the history is to use the `context.ContinueAsNew(input)` method, as described in the article I linked to previously. Basically, you could change your orchestration to use that instead of looping, and include any necessary loop state in the input payload.

If you are going to make this change, you'll want to do it for a new instance on a new task hub (see our [Versioning](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-versioning) documentation). The existing instance likely cannot be salvaged unless you temporarily run your app on a VM with a larger memory limit, let it complete, and then move your app back to the smaller, Consumption instances.

**Comment by Haletski** [comment]
@cgillum Thanks for your assistance. It absolutely makes sense for me.
